File-Storage-Security/Deployment/aws-python-deploy-to-all-existing-s3/deploy.py

```diff
@@ -106,49 +112,38 @@ def get_encryption_region(list_of_buckets):
                         print("S3: " + bucket_name + " has tag FSSMonitored == no; skipping")
                         break
                     elif tags["Key"] == "FSSMonitored" and tags["Value"].lower() == "yes":
-                        print("S3: "+ bucket_name + " FSS Tag Found!, FSS is already deployed!")
+                        print("S3: " + bucket_name + " FSS Tag Found!, FSS is already deployed!")
                         break
 
         except ClientError:
             no_tags = "does not have tags"
             tag_status = no_tags
-            add_tag(s3_client, bucket_name, tag_list=[])
             deploy_storage(kms_arn, region, bucket_name)
+            add_tag(s3_client, bucket_name, tag_list=[])
+
 
 def add_tag(s3_client, bucket_name, tag_list):
-    tag_list.append({'Key':'FSSMonitored', 'Value': 'Yes'})    
+    tag_list.append({'Key': 'FSSMonitored', 'Value': 'Yes'})
     s3_client.put_bucket_tagging(
         Bucket=bucket_name,
         Tagging={"TagSet": tag_list},
     )
-    
-#function to deploy fss storage stack
+
+
+# function to deploy fss storage stack
 def deploy_storage(kms_arn, region, bucket_name):
-    #set up aws Config for region changes
+    # set up aws Config for region changes
     print("deploying to : " + bucket_name)
     my_region_config = Config(
-        region_name = region,
-        signature_version = 'v4',
-        retries = {
+        region_name=region,
+        signature_version='v4',
+        retries={
             'max_attempts': 10,
             'mode': 'standard'
         }
     )
-    # gather cloud one ext id
-    r = http.request(
-        "GET",
-        stacks_api_url+"external-id",
-        headers={
-            "Authorization": "ApiKey " + ws_api,
-            "Api-Version": "v1",
-        },
-    )
-    try:
-        ext_id = json.loads(r.data.decode("utf-8"))['externalID']
-    except json.decoder.JSONDecodeError:
-        time.sleep(1)
-        ext_id = json.loads(r.data.decode("utf-8"))['externalID']
```

Comment:
The customer received an error when running the script. We figured it could be added as an argument to avoid that.

Reply:
When ran I found 3 errors actually occurring.
1. If the exclusions.txt had any spaces it would cause the script to err out. 
2. If the bucket had an existing s3 event notification in use it would err out.
3. It would tag the bucket before it deploy protection